qiita_ware/ebi.py

```diff
@@ -885,8 +866,14 @@ def generate_demultiplexed_fastq(self, rewrite_fastq=False, mtime=None):
         """
         ppd = self.preprocessed_data
 
-        if not isdir(self.ebi_dir) or rewrite_fastq:
-            makedirs(self.ebi_dir)
+        dir_not_exists = not isdir(self.full_ebi_dir)
+        if dir_not_exists or rewrite_fastq:
+            if dir_not_exists:
+                makedirs(self.full_ebi_dir)
+                filetype_id = convert_to_id('directory', 'filepath_type')
+                insert_filepaths([(self.full_ebi_dir, filetype_id)], -1,
```

Comment:
My only concern with this solution is that if we call "purge_filepaths" at some point, these entries in the filepath table will be removed because they're not linked from any other table